### PR TITLE
Clean Clippy warnings

### PR DESCRIPTION
**Pull request by gkorland**
* clean pedantic warnings

### CODE REVIEW

File: src/number.rs [hunk 1]
```diff
@@ -88,12 +88,12 @@ impl Header {
...
             match self.type_ {
-                NumberType::Static => Some(self.static_ as i64),
-                NumberType::I24 => Some(self.as_i24_unchecked() as i64),
+                NumberType::Static => Some(i64::from(self.static_)),
+                NumberType::I24 => Some(i64::from(self.as_i24_unchecked())),
                 NumberType::I64 => Some(*self.as_i64_unchecked()),
                 NumberType::U64 => {
                     let v = *self.as_u64_unchecked();
-                    if v <= i64::MAX as u64 {
+                    if i64::try_from(v).is_ok() {
```
Comment (Diggsey) [inline src/number.rs:96]: This `if` is now redundant.

File: src/number.rs [hunk 2]
```diff
@@ -443,7 +445,7 @@ impl INumber {
     }
 
     fn new_u64(value: u64) -> Self {
-        if value <= i64::MAX as u64 {
+        if i64::try_from(value).is_ok() {
```
Comment (Diggsey) [inline src/number.rs:448]: This should be `if let`

File: src/object.rs [hunk 3]
```diff
@@ -327,11 +327,13 @@ impl<'a> OccupiedEntry<'a> {
         }
     }
     /// Returns a reference to the key at this entry
+    #[must_use]
     pub fn key(&self) -> &IString {
         self.get_key_value().0
     }
 
     /// Removes and returns the entry as a (key, value) pair.
+    #[must_use]
```
Comment (Diggsey) [inline src/object.rs:336]: These `#[must_use]` annotations seem over-zealous. For example, the corresponding method in `libstd` does not have `#[must_use]`. Which clippy lint is requiring this?

**Review by Diggsey — CHANGES_REQUESTED** [review]

File: src/object.rs (hunk 3 above)
Comment (gkorland) [reply src/object.rs:336]: https://rust-lang.github.io/rust-clippy/master/index.html#must_use_candidate

Comment (Diggsey) [reply src/object.rs:336]: OK, I think this is a false positive though. `remove_entry` mutates the underlying map and should not be marked `must_use`.

File: src/object.rs [hunk 4]
```diff
@@ -359,6 +363,7 @@ impl<'a> OccupiedEntry<'a> {
     }
 
     /// Removes this entry and returns its value.
+    #[must_use]
```
Comment (Diggsey) [inline src/object.rs:366]: This also should not be `#[must_use]`

**Review by Diggsey — CHANGES_REQUESTED** [review]

**Comment by gkorland** [comment]
@Diggsey fixed format

**Comment by Diggsey** [comment]
Thanks!

**Pull request merged**